source/Calamari.Common/Features/StructuredVariables/YamlFormatVariableReplacer.cs

```diff
@@ -7,11 +14,60 @@ public interface IYamlFormatVariableReplacer : IFileFormatVariableReplacer
     public class YamlFormatVariableReplacer : IYamlFormatVariableReplacer
     {
         public string FileFormatName => "YAML";
-        
+
         public bool TryModifyFile(string filePath, IVariables variables)
         {
-            // Yaml not yet supported.
-            return false;
+            var variablesByKey = variables
+                .DistinctBy(v => v.Key)
+                .ToDictionary(v => v.Key, v => v.Value);
+
+            // Read and transform the input file
+            var outputEvents = new List<ParsingEvent>();
+            try
+            {
+                using (var reader = new StreamReader(filePath))
+                {
+                    var parser = new Parser(reader);
+                    var classifier = new YamlEventStreamClassifier();
+                    while (parser.MoveNext())
+                    {
+                        var ev = parser.Current;
+                        if (ev == null)
+                            continue;
+
+                        var found = classifier.Process(ev);
+                        if (found is YamlScalarValueNode scalar
+                            && variablesByKey.TryGetValue(scalar.Path, out string newValue))
+                        {
+                            ev = scalar.ReplaceValue(newValue);
+                        }
+
+                        outputEvents.Add(ev);
+                    }
+                }
+            }
+            catch (SyntaxErrorException)
+            {
+                // TODO: Report where the problem was in the input file
```

Comment:
Good point here, and I think another piece of evidence that having the user specify the file format is a good idea. Take this example:

1. User creates a package with a .properties file
2. User tries to deploy package
3. Calamari tries to parse the file as json, and shows an exception with details of the syntax error
4. Calamari tries to parse the file as yaml, and shows an exception with details of the syntax error
5. Calamari tries to parse the file as xml, and shows an exception with details of the syntax error
6. Calamari tries to parse the file as .properties, which succeeds
7. The deployment finishes.
8. User looks in logs to see 3 parsing errors even though the parsing succeeded.

If we know what the file _should_ be, then we can happily log the syntax error here and make life easy for the customer. But if we have to guess, we need to be a little more nuanced in how we handle syntax errors, and when to display them.

Reply:
We could show just one in cases where we're "confident" (such as a file extension match). But more and more I'm thinking that by the time a user enters a filename, correcting the file type isn't much effort, particularly if we auto-select the most likely one as soon as they enter the filename.